xonsh/foreign_shells.py

```diff
@@ -173,6 +177,11 @@ def foreign_shell_data(shell, interactive=True, login=False, envcmd=None,
     funcscmd = DEFAULT_FUNCSCMDS.get(shell, 'echo {}') if funcscmd is None else funcscmd
     command = COMMAND.format(envcmd=envcmd, aliascmd=aliascmd, prevcmd=prevcmd,
                              postcmd=postcmd, funcscmd=funcscmd).strip()
+    if shell == 'cmd':
+        cmd.remove('-c')
+        cmd.append('/C')
+        command = command.replace('\n','&&')
+
```

Comment:
As for the replace, I think that writing out the command to a named temporary file and then passing in the file name is a better option than replacing newlines with `&&`. This ability can be controled by a keyword argument and doesn't have be `cmd` specific (though cmd will us it).